cactus/utils/internal.py

```diff
@@ -1,12 +1,51 @@
 #coding:utf-8
+import contextlib
 import six
 import inspect
 
 # Adapted from: http://kbyanc.blogspot.com/2007/07/python-more-generic-getargspec.html
 
-
 FUNC_OBJ_ATTR = "__func__" if six.PY3 else "im_func"
 
+# The alternate for inspect.ArgSpec which was deprecated
+class ArgumentInfo:
+    def __init__(self, args, varargs, varkw, defaults):
+        self.args = args
+        self.varargs = varargs
+        self.varkw = varkw
+        self.defaults = defaults
+
+# format the spec to required format of ArgumentInfo
+def get_argument_info(spec):
+    spec_values = spec.parameters.values()
+    args = [param.name for param in spec_values]
+    varargs = None
+    varkw = None
+    defaults = [
```

Comment:
I think it was because, I updated as such that the output format resembles the previous code so it does not affect other code that depend on the output as I don't know what the other codes might be. So I kept the previous structure same

Reply:
So, what do I actually need to remove

Reply:
Rewrite the code using one cycle instead two.

Reply:
done